services/src/pro/datasets/external/sentinel_s2_l2a_cogs.rs

```diff
@@ -247,18 +247,37 @@ impl SentinelS2L2aCogsMetaData {
 
         features.sort_by_key(|a| a.properties.datetime);
 
+        let mut start_times: Vec<TimeInstance> = Vec::with_capacity(features.len());
+        for (i, feature) in features.iter().enumerate() {
+            let start = feature.properties.datetime;
+            let real_start = if i == 0 {
+                TimeInstance::from(start)
+            } else {
+                let prev_start = features[i - 1].properties.datetime;
+                if start == prev_start {
+                    log::debug!("duplicate start time: {}", start);
+                    start_times[i - 1] + 1
+                } else {
+                    TimeInstance::from(start)
+                }
+            };
+
+            start_times.push(real_start);
+        }
```

Comment:
Please write a test for that. This won't work if you have three consecutive timestamps
`[1, 1, 1]`, then your algorithm would produce `[1, 2, 1]` I guess.

Reply:
[a87638a](https://github.com/geo-engine/geoengine/pull/480/commits/a87638a2c5618301fb26ea4990bfdcf79f790006)